src/app/calculator/furnaces/flue-gas/flue-gas-modal/flue-gas-modal.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { SettingsDbService } from '../../../../indexedDb/settings-db.service';
+import { FlueGas, FlueGasOutput } from '../../../../shared/models/phast/losses/flueGas';
+import { Settings } from '../../../../shared/models/settings';
+import { FlueGasService } from '../flue-gas.service';
+
+@Component({
+  selector: 'app-flue-gas-modal',
+  templateUrl: './flue-gas-modal.component.html',
+  styleUrls: ['./flue-gas-modal.component.css']
+})
+export class FlueGasModalComponent implements OnInit {
+
+  @Output('closeModal')
+  closeModal = new EventEmitter<number>();
+  @Output('hideModal')
+  hideModal = new EventEmitter();
+  @Input()
+  settings: Settings;
+  
+  method: string = 'By Mass';
+  baselineData: FlueGas;
+  modificationData: FlueGas;
+  baselineDataSub: Subscription;
+  modificationDataSub: Subscription;
+  outputSubscription: Subscription;
+  output: FlueGasOutput;
+  
+  constructor(private settingsDbService: SettingsDbService, 
+              private flueGasService: FlueGasService) { }
+
+  ngOnInit(): void {
+    if (!this.settings) {
+      this.settings = this.settingsDbService.globalSettings;
+    }
+
+    let existingInputs = this.flueGasService.baselineData.getValue();
+    if(!existingInputs) {
+      this.flueGasService.initDefaultEmptyOutput();
+      this.flueGasService.initDefaultEmptyInputs();
+    }
+    this.output = this.flueGasService.output.getValue();
```

Comment:
Have output set here and in the subscriptions. Probably just use in the subscription